stingray/largememory.py

```diff
@@ -556,9 +559,8 @@ def _addSpectra(final_spectra, curr_spec, flag):
 
         return final_spectra
 
-    np.multiply(np.add(final_spectra.freq, curr_spec.freq),
-                curr_spec.m,
-                out=final_spectra.freq)
+    assert np.allclose(final_spectra.freq, curr_spec.freq), \
+        "Summing a spectrum with incompatible frequency values"
```

Comment:
Why is this needed here?

Reply:
Ok, I see. So, what's the meaning of `flag`?

Reply:
The flag is used to indicate the first iteration. As some processing will be different in the first iteration than other iterations

Reply:
Maybe call it `first_iteration` instead :)?

Reply:
Okay will do. Descriptive names 💯